packages/api/src/apps/lsp_api/app.ts

```diff
@@ -131,7 +131,9 @@ export default async (env: ProcessEnv) => {
     // these services can optionally be configured with a config object, but currently they are undefined or have defaults
     blocks: Services.Blocks(undefined, appState),
     emps: Services.EmpState({ debug }, appState),
-    registry: await Services.Registry({ debug }, appState, (data) => serviceEvents.emit("empRegistry", data)),
+    registry: await Services.Registry({ debug }, appState, (event, data) =>
+      serviceEvents.emit("empRegistry", event, data)
```

Comment:
Are event and data different types? Surprised typescript didn't help here

Reply:
yea that is actually really wierd. typescript should have saw that, will look into it

Reply:
for some reason its treating functional arguments as optional, even though they should be defined:
`(event: Events, data: EmitData) => void` is the callback signature, but i can just pass in `()=>{..etc}` and typescript does not complain

Reply:
https://github.com/Microsoft/TypeScript/wiki/FAQ#why-are-functions-with-fewer-parameters-assignable-to-functions-that-take-more-parameters